search-app/src/cmr/search/services/parameters/converters/shapefile.clj

```diff
@@ -0,0 +1,146 @@
+    [cmr.common.log :refer (debug)]
+    [cmr.common.mime-types :as mt]
+    [cmr.common-app.services.search.group-query-conditions :as gc]
+    [cmr.common-app.services.search.params :as p]
+    [cmr.common.services.errors :as errors]
+    [cmr.search.models.query :as qm]
+    [cmr.search.services.parameters.converters.geometry :as geo]
+    [cmr.common.util :as util])
+  (:import
+   (java.io BufferedInputStream File FileReader FileOutputStream FileInputStream)
+   (java.nio.file Files)
+   (java.nio.file.attribute FileAttribute)
+   (java.net URL)
+   (java.util HashMap)
+   (java.util.zip ZipFile ZipInputStream)
+   (org.apache.commons.io FilenameUtils)
+   (org.geotools.data DataStoreFinder FileDataStoreFinder Query)
+   (org.geotools.data.simple SimpleFeatureSource)
+   (org.geotools.data.geojson GeoJSONDataStore)
+   (org.geotools.geometry.jts JTS)
+   (org.geotools.referencing CRS)
+   (org.geotools.util URLs)))
+
+(def EPSG-4326-CRS (CRS/decode "EPSG:4326" true))
+
+(defconfig enable-shapefile-parameter-flag
+  "Flag that indicates if we allow spatial searching by shapefile."
+  {:default true :type Boolean})
+
+(defn- unzip-file
+  "Unzip a file (of type File) into a temporary directory and return the directory path as a File"
+  [source]
+  (try
+    (let [target-dir (Files/createTempDirectory "Shapes" (into-array FileAttribute []))]
+      (with-open [zip (ZipFile. source)]
+        (let [entries (enumeration-seq (.entries zip))
+              target-file #(File. (.toString target-dir) (str %))]
+          (doseq [entry entries :when (not (.isDirectory ^java.util.zip.ZipEntry entry))
+                  :let [f (target-file entry)]]
+            (debug (format "Zip file entry: [%s]" (.getName entry)))
+            (io/copy (.getInputStream zip entry) f))))
+      (.toFile target-dir))
+    (catch Exception e
+      (errors/throw-service-error :bad-request (str "Error while uncompressing zip file: " (.getMessage e))))))
+
+(defn find-shp-file
+  "Find the .shp file in the given directory (File) and return it as a File"
+  [dir]
+  (let [files (file-seq dir)]
+    (first (filter #(= "shp" (FilenameUtils/getExtension (.getAbsolutePath %))) files))))
+
+(defn geometry->conditions
+  "Get one or more conditions for the given Geometry. This will only
+  return more than one condition if the Geometry is a GeometryCollection."
+  [geometry]
+  (let [num-geometries (.getNumGeometries geometry)]
+    (for [index (range 0 num-geometries)
+          :let [sub-geometry (.getGeometryN geometry index)]]
+      (geo/geometry->condition sub-geometry))))
+
+(defn transform-to-epsg-4326
+  "Transform the geometry to WGS84 CRS if is not already"
+  [geometry src-crs]
+  ;; if the source CRS is defined and not already WGS84 then transform the geometry to WGS84
+  (if (and src-crs
+           (not (= (.getName src-crs) (.getName EPSG-4326-CRS))))
+    (let [src-crs-name (.getName src-crs)
+          _ (debug (format "Source CRS: [%s]" src-crs-name))
```

Comment:
You could move these debugs out of the let by changing lines 78-80 to:
```
(if-let [transform (CRS/findMathTransform src-crs EPSG-4326-CRS false)
...
```

Reply:
Done